src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -406,16 +406,16 @@ public Uri(Uri baseUri, string relativeUri)
         //
         protected Uri(SerializationInfo serializationInfo, StreamingContext streamingContext)
         {
-            string uriString = serializationInfo.GetString("AbsoluteUri"); // Do not rename (binary serialization)
+            string? uriString = serializationInfo.GetString("AbsoluteUri"); // Do not rename (binary serialization)
 
-            if (uriString.Length != 0)
+            if (uriString!.Length != 0)
             {
                 CreateThis(uriString, false, UriKind.Absolute);
                 return;
             }
 
             uriString = serializationInfo.GetString("RelativeUri");  // Do not rename (binary serialization)
-            if ((object)uriString == null)
+            if ((object?)uriString == null)
```

Comment:
These days there's no benefit to casting to object before the `== null` (to avoid the `op_Equality` call). Why not just fix-up this old pattern?

Either:

```c#
if (uriString == null)
```

...or:

```c#
if (uriString is null)
```

?

Applies throughout.

Reply:
Same.